src/main/java/core/basesyntax/MyLinkedList.java

```diff
@@ -3,45 +3,166 @@
 import java.util.List;
 
 public class MyLinkedList<T> implements MyLinkedListInterface<T> {
+    private int size = 0;
+    private Node<T> first;
+    private Node<T> last;
+
+    private static class Node<T> {
+        private T value;
+        private Node<T> prev;
+        private Node<T> next;
+
+        public Node(Node<T> prev, T value, Node<T> next) {
+            this.prev = prev;
+            this.value = value;
+            this.next = next;
+        }
+    }
+
+    public void checkOutIndex(int index) {
+        if (index < 0 || index >= size) {
+            throw new IndexOutOfBoundsException("Wrong index");
+        }
+    }
+
     @Override
     public void add(T value) {
+        Node<T> node = new Node<>(last, value, null);
+        Node<T> oldNode = last;
+        last = node;
+        if (size == 0) {
+            first = node;
+        } else {
+            oldNode.next = node;
+        }
+        size++;
     }
 
     @Override
     public void add(T value, int index) {
+        if (size == 0 && index == 0) {
+            Node<T> node = new Node<>(null, value, null);
+            first = last = node;
+            size++;
+            return;
+        }
+        if (index == 0) {
+            Node<T> node = new Node<>(null, value, first);
+            Node<T> oldNode = first;
+            first = oldNode.prev = node;
+            size++;
+            return;
+        }
+        if (index == size) {
+            Node<T> node = new Node<>(last, value, null);
+            Node<T> oldNode = last;
+            last = oldNode.next = node;
+            size++;
+            return;
+        }
+        checkOutIndex(index);
+        Node<T> oldNode = first;
+        int i = 0;
+        while (i < index) {
+            oldNode = oldNode.next;
+            i++;
+        }
+        Node<T> node = new Node<>(oldNode.prev, value, oldNode);
+        oldNode.prev.next = oldNode.prev = node;
+        size++;
     }
 
     @Override
     public void addAll(List<T> list) {
+        int i = 0;
+        while (i < list.size()) {
+            add(list.get(i));
+            i++;
+        }
     }
 
     @Override
     public T get(int index) {
-        return null;
+        checkOutIndex(index);
+        Node<T> getNode = first;
+        int i = 0;
+        while (i < index) {
+            getNode = getNode.next;
+            i++;
+        }
+        return getNode.value;
```

Comment:
let's create a separate method that will be looking for a node by index because you have in more than one method and use for loop